Apsitvarkom.UnitTests/DataAccess/GoogleGeocoderTests.cs

```diff
@@ -1,48 +1,88 @@
 using System.Net;
 using Apsitvarkom.Configuration;
 using Apsitvarkom.DataAccess;
+using Apsitvarkom.Models;
 using Moq;
 using Moq.Protected;
 
 namespace Apsitvarkom.UnitTests.DataAccess;
 
 public class GoogleGeocoderTests
 {
-    private static readonly IApiKeyProvider _apiKeyProvider = new Mock<IApiKeyProvider>().Object;
+    private Mock<IApiKeyProvider> _apiKeyProvider;
+    private Mock<HttpMessageHandler> _handlerMock;
+    private IGeocoder _geocoder;
+
+    [SetUp]
+    public void SetUp()
+    {
+        _apiKeyProvider = new Mock<IApiKeyProvider>();
+        _handlerMock = new Mock<HttpMessageHandler>();
+        var httpClient = new HttpClient(_handlerMock.Object)
+        {
+            BaseAddress = new Uri("https://mockreversegeocoder.com")
+        };
+        _geocoder = new GoogleGeocoder(httpClient, _apiKeyProvider.Object);
+    }
+
+    [Test]
+    public Task OkStatusCodeReturned_SingleAddressRetrieved_TitleSuccessfullyReturned()
+    {
+        const string title = "Hello World";
+        const string responseJsonString = "{\"results\":[{\"formatted_address\":\"" + title + "\"}], \"status\": \"OK\"}";
+
+        return TestReverseGeocodeRequest(responseJsonString, title);
+    }
 
     [Test]
-    public async Task SuccessfullyFindsFirstAddress()
+    public Task OkStatusCodeReturned_SeveralAddressesRetrieved_FirstTitleReturned()
     {
         const string title = "Hello World";
-        const string responseJsonString = "{\"results\":[{\"formatted_address\":\"" + title + "\"}, {\"formatted_address\": \"second address\"}]}";
+        const string responseJsonString = "{\"results\":[{\"formatted_address\":\"" + title + "\"}, {\"formatted_address\": \"second address\"}], \"status\": \"OK\"}";
 
-        var handlerMock = new Mock<HttpMessageHandler>();
+        return TestReverseGeocodeRequest(responseJsonString, title);
+    }
+
+    [Test]
+    public Task OkStatusCodeReturned_NullAddressRetrieved_NullReturned()
+    {
+        const string responseJsonString = "{\"results\":[{\"formatted_address\":null}], \"status\": \"OK\"}";
```

Comment:
I am not sure what adding `status` to the json result string gives us here. I've read [the docs](https://developers.google.com/maps/documentation/geocoding/requests-reverse-geocoding#reverse-status-codes) but that doesn't give us anything since we don't even take into account that field (in `GoogleGeocoder.cs` we only deserialize `results` -> `formatted_address`). You could write any other valid json and we would just ignore it just like we now ignore that `status` you put in the tests.

Reply:
Practically for our use case it doesn't add nothing, good point. I have considered slightly altering what we parse from the response to include status too, but figured it would probably be redundant. I guess I just didn't remove it afterwards. Will delete